drivers/SmartThings/harman-luxury/src/api/nsdk.lua

```diff
@@ -0,0 +1,246 @@
+local cosock = require "cosock"
+local http = cosock.asyncify "socket.http"
+local ltn12 = require "ltn12"
+local url = require "net.url"
+local json = require "st.json"
```

Comment:
`json.decode` can throw an `error` on malformed input, which can crash the whole driver.

This API needs to be either documented to note this, or cleaned up to avoid this. In the meantime, I would strongly suggest using `pcall` everywhere that you call `json.decode`